.circleci/config.yml

```diff
@@ -75,106 +66,327 @@ commands:
 
-  mbc:
+  angio-deploy-only:
+    executor:
+      name: commit-executor
+    steps:
+      - deploy-stored-build: &angio-params
+          study_key: angio
+          study_guid: ANGIO
+
+  mbc-deploy-only:
     executor:
       name: commit-executor
     steps:
-      - build-and-deploy-study:
+      - deploy-stored-build: &mbc-params
           study_key: mbc
           study_guid: cmi-mbc
 
+  osteo-build-and-deploy:
+    executor:
+      name: commit-executor
+    steps:
+      - app-build-and-deploy:
+          <<: *osteo-params
+
+  angio-build-and-deploy:
+    executor:
+      name: commit-executor
+    steps:
+      - app-build-and-deploy:
+          <<: *angio-params
+
+  brain-build-and-deploy:
+    executor:
+      name: commit-executor
+    steps:
+      - app-build-and-deploy:
+          <<: *brain-params
+
+  mbc-build-and-deploy:
+    executor:
+      name: commit-executor
+    steps:
+      - app-build-and-deploy:
+          <<: *mbc-params
+
+parameters:
+  study_key:
+    type: string
+    default: "UNKNOWN"
+  do-builds:
+    type: boolean
+    default: false
+
 workflows:
   version: 2
-  build-deploy:
+
+  build-and-deploy-all-apps-workflow:
     jobs:
-      - osteo:
-          # todo: more DRY on branch filtering
-          filters:
-            branches:
-              only:
-                - develop
-      - angio:
+      - osteo-build-and-deploy: &build-and-deploy-filters
           filters:
             branches:
               only:
                 - develop
-      - brain:
+      - brain-build-and-deploy:
+          <<: *build-and-deploy-filters
+      - angio-build-and-deploy:
+          <<: *build-and-deploy-filters
+      - mbc-build-and-deploy:
+          <<: *build-and-deploy-filters
+
+  launch-all-app-builds-workflow:
+    # using do-builds param to select which of the two workflows using the build-on-tag-filters
+    # should run; should be this one that just launches the builds or the one actually doing the builds
+    unless: << pipeline.parameters.do-builds >>
+    jobs:
+      - conditionally-launch-build-and-store-all-job: &build-and-store-filters
           filters:
             branches:
               only:
-                - develop
-      - mbc:
+                - /^rc.*/
+                - /^hotfix.*/
+
+  build-app-workflow:
+    when: << pipeline.parameters.do-builds >>
+    jobs:
+      - app-build-and-store-job:
+          <<: *build-and-store-filters
+
+  deploy-all-apps-workflow:
+    jobs:
+      - osteo-deploy-only: &deploy-filters
           filters:
             branches:
               only:
-                - develop
-  nightly:
-    # run a build on develop using --prod --aot as an early warning for prod/aot issues
+                - test
+                - staging
+                - production
+      - brain-deploy-only:
+          <<: *deploy-filters
+      - mbc-deploy-only:
+          <<: *deploy-filters
+      - angio-deploy-only:
+          <<: *deploy-filters
 
```

Comment:
This is the workflow for deployments of saved builds. One thing to notice is that the build-only workflow is detached from the deploy-only workflow. So if:
a) the process of creating the rc_* or hotfix_* branch is not followed.
b) You move PUSH to test or staging a few seconds after having pushed to rc_* or hotfix_*
The deploy will fail because the archive will not exist yet.